otherlibs/stdune/src/bit_set.ml

```diff
@@ -0,0 +1,41 @@
+module Common = struct
```

Comment:
It's not clear to me what principle was used to factor out these functions into a separate module. For example, why is `let singleton x = 1 lsl Element.to_int x` not part of it? Should it be?

Reply:
(Same question for `type t = int`.)

Reply:
The principle is that all of those implementations can be shared between all instantiations of the functor. In other words, they're independent of the representation of the elements. It's a small optimization that probably doesn't matter.

Reply:
I see. I'm thinking that a part of `singleton` (that does `1 lsl`) can still be moved into `Common` though maybe that doesn't matter. Maybe add a comment explaining your principle so that others aren't wondering in the same way as me.

Reply:
> I'm thinking that a part of singleton (that does 1 lsl) can still be moved into Common though maybe that doesn't matter.

Right, but that function depends on the representation of elements to be used correctly. So it's not too useful.

> Maybe add a comment explaining your principle so that others aren't wondering in the same way as me.

Will do